core/shared/src/main/scala/fs2/async/async.scala

```diff
@@ -18,8 +19,8 @@ package object async {
     mutable.Signal(initialValue)
 
   /** Creates a `[[mutable.Semaphore]]`, initialized to the given count. */
-  def semaphore[F[_]: Concurrent](initialCount: Long): F[mutable.Semaphore[F]] =
-    mutable.Semaphore(initialCount)
+  def semaphore[F[_]: Concurrent](initialCount: Long): F[Semaphore[F]] =
```

Comment:
Think we still need these constructors here?

Reply:
I wasn't too sure, I kinda liked having them as they made things easier to migrate, and in general I like the fact they you can save some imports, e.g. call `refOf` without having to import `Ref`.
Not too hung up on it one way or another.

I was also thinking of making a small PR to cats-effect to rename `modifyAndReturn` to `modify` and `modify` to `modify_`, as you can see `modifyAndReturn` is needed pretty much everywhere, and `modify_` is consistent with `traverse_` and co (throw away the result). WDYT? (very minor, cosmetic issue)

Reply:
I lean towards removing the aliases as I don't like introducing multiple ways to do the same thing. I think if folks are using `Ref` and `Deferred` and `Semaphore`, an extra import is okay.

No strong opinion on the rename to `modify_`.

Reply:
BTW, if you want to do the rename, let's get that in before the M2 release, which I think is tomorrow.

Reply:
Cool, I'll remove the aliases then :)

And I'll ask about the rename to see what people think, and if positive, I'll open a PR for that